### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by arulelango-moj-digital**
This PR adds the functionality for the submission of the Request A Link page to call the send-legal-mail-api to request the magic link is sent

### CODE REVIEW

File: integration_tests/mockApis/sendLegalMail.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { SuperAgentRequest } from 'superagent'
+import { stubFor } from './wiremock'
+
+const stubRequestLink = (): SuperAgentRequest =>
+  stubFor({
+    request: {
+      method: 'POST',
+      urlPattern: '/send-legal-mail/link/email',
+      bodyPatterns: [{ matchesJsonPath: '$[?(@.email =~ /.*/i && @.sessionID =~ /.*/i)]' }],
```
Comment (mikehalmamoj) [inline integration_tests/mockApis/sendLegalMail.ts:9]: We're not actually going to send the session ID as the link is not tied to the session - we tried that in the PoC but then decided not to, but I didn't mention it on this ticket (my bad!)

File: server/routes/link/RequestLinkController.ts [hunk 2]
```diff
@@ -16,10 +19,18 @@ export default class RequestLinkController {
       return res.redirect('request-link')
     }
 
-    const emailSentTo = req.session.requestLinkForm.email
-    delete req.session.requestLinkForm
+    return this.magicLinkService
+      .requestLink(req.session.requestLinkForm.email, req.sessionID)
```
Comment (mikehalmamoj) [inline server/routes/link/RequestLinkController.ts:23]: As per other comment, session ID not required.

**Review by mikehalmamoj — DISMISSED** [review]
Just need session ID removing from the API call but can do that in the next PR.

File: helm_deploy/send-legal-mail-to-prisons/values.yaml [hunk 3]
```diff
@@ -38,6 +38,8 @@ generic-service:
       APPINSIGHTS_INSTRUMENTATIONKEY: "APPINSIGHTS_INSTRUMENTATIONKEY"
       API_CLIENT_ID: "API_CLIENT_ID"
       API_CLIENT_SECRET: "API_CLIENT_SECRET"
+      SYSTEM_CLIENT_ID: "SYSTEM_CLIENT_ID"
+      SYSTEM_CLIENT_SECRET: "SYSTEM_CLIENT_SECRET"
```
Comment (mikehalmamoj) [inline helm_deploy/send-legal-mail-to-prisons/values.yaml:42]: Ah! We'd better get this secret created before merging this PR or the delpoyment will fail.

File: integration_tests/mockApis/sendLegalMail.ts (hunk 1 above)
Comment (nathanrussell-moj-digital) [reply integration_tests/mockApis/sendLegalMail.ts]: I've removed sessionId 👍

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**